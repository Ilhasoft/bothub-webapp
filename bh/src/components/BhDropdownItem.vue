<template>
  <div
    class="bh-dropdown-item"
    @click="$emit('click', $event)"
  >
    <slot />
  </div>
</template>

<script>
export default {
  name: 'BhDropdownItem',
};
</script>

<style lang="scss">
@import '@scss/colors.scss';


.bh {
  &-dropdown-item {
    display: flex;
    padding: .5rem .75rem;
    cursor: pointer;

    &:hover {
      color: $color-fake-white;
      background-color: $color-primary;
    }

    & > * {
      margin-right: 8px;

      &:last-child {
        margin-right: 0;
      }
    }
  }
}
</style>
