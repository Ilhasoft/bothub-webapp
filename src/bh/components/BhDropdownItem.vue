<template>
  <div
    class="bh-dropdown-item"
    @click="$emit('click', $event)">
    <slot />
  </div>
</template>

<script>
export default {
  name: 'BhDropdownItem',
};
</script>

<style lang="scss">
@import '~bh/assets/scss/colors.scss';
@import '~bh/assets/scss/typography.scss';

.bh {
  &-dropdown-item {
    @include typography();

    display: flex;
    padding: .5rem .75rem;
    cursor: pointer;

    &:hover {
      color: $color-primary;
      background-color: $color-lighter-grey;
    }

    & > * {
      margin-right: 8px;

      &:last-child {
        margin-right: 0;
      }
    }
  }
}
</style>
