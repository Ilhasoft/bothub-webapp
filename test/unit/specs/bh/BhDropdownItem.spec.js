import { shallowMount, createLocalVue } from '@vue/test-utils';

import BH from 'bh';
import BhDropdownItem from 'bh/components/BhDropdownItem';


const localVue = createLocalVue();
localVue.use(BH);

describe('BhDropdownItem.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(BhDropdownItem, {
      localVue,
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
