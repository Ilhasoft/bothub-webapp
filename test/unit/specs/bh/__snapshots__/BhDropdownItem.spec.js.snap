// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BhDropdownItem.vue renders correctly 1`] = `<div class="bh-dropdown-item"></div>`;
